### PR TITLE
Build: Rebuild package stylesheet entrypoints for file updates

### PR DESCRIPTION
**Pull request by aduth**
Regression introduced in #15230

This pull request seeks to resolve an issue with the revised build implementation merged in #15230 . Stylesheet rebuilds are always intended to compile from the _entrypoints_, not from the individual files which have been updated. This is distinct from JavaScript files, which are built individually and which rely on Webpack to manage entrypoint bundling.

**Implementation Notes:**

Since this was already stream-based, I implemented the fix using a stream transform to map the files modified to the equivalent entrypoints.

**Testing Instructions:**

Run the command:

```
npm run clean:packages && node ./bin/packages/build.js packages/block-library/src/columns/editor.scss packages/block-library/src/columns/style.scss packages/block-library/src/columns/edit.js
```

Verify at `./packages/block-library`:

- `build` includes `columns/edit.js` and `columns/edit.js.map`
- `build-style` includes `editor.css`, `style.css`, `theme.css` (and RTL counterparts)

The fix is specifically in the latter point. Previously, the build script would output the compiled CSS at `build-style/columns/editor.css` and `build-style/columns/style.css`. The former point is merely to assure there is no effect on JavaScript build.

### CODE REVIEW

File: bin/packages/build.js [hunk 1]
```diff
@@ -18,6 +18,52 @@ const files = process.argv.slice( 2 );
...
+ *
+ * @return {Transform} Stream transform instance.
+ */
+function createStyleEntryTransform() {
+	const packages = new Set;
+
+	return new Transform( {
+		objectMode: true,
+		async transform( file, encoding, callback ) {
+			// Only stylesheets are subject to this transform.
```
Comment (aduth) [inline bin/packages/build.js]: I thought to be clever / functional here with some sort of conditional flow (an [`overEvery`](https://lodash.com/docs/4.17.11#overEvery) and/or series `Promise`) but I couldn't come up with an implementation which wouldn't end up being unnecessarily obscure / unreadable.

Comment (noisysocks) [reply bin/packages/build.js]: Best to keep it simple! 😀

**Review by noisysocks — APPROVED** [review]
Tested this by following your instructions and by making a few CSS changes while running `npm run dev`. Took me a little while to grok how stream transforms work but got there in the end 👍

**Comment by gziolo** [comment]
I better merge it to ensure `npm run dev` works seamlessly. I trust you both :)

**Pull request merged**